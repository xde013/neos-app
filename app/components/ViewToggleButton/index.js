/**
 *
 * ViewToggleButton
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import ToggleButton from '@material-ui/lab/ToggleButton';
import TableChartOutlinedIcon from '@material-ui/icons/TableChartOutlined';
import InsertChartOutlinedOutlinedIcon from '@material-ui/icons/InsertChartOutlinedOutlined';
import Tooltip from '@material-ui/core/Tooltip';

const ViewToggleButton = ({ toggleHandler }) => {
  const [selected, setSelected] = React.useState(false);

  return (
    <Tooltip title={`Toggle to ${selected ? 'Chart' : 'Table'} View`}>
      <ToggleButton
        value={false}
        selected
        onChange={() => {
          setSelected(!selected);
          toggleHandler(!selected);
        }}
        color="primary"
      >
        {selected ? (
          <InsertChartOutlinedOutlinedIcon fontSize="large" />
        ) : (
          <TableChartOutlinedIcon fontSize="large" />
        )}
      </ToggleButton>
    </Tooltip>
  );
};

ViewToggleButton.propTypes = {
  toggleHandler: PropTypes.func.isRequired,
};

export default ViewToggleButton;
