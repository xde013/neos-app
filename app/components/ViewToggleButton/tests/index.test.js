/**
 *
 * Tests for ViewToggleButton
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import ViewToggleButton from '../index';

describe('<ViewToggleButton />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<ViewToggleButton toggleHandler={() => {}} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<ViewToggleButton toggleHandler={() => {}} />);
    expect(firstChild).toMatchSnapshot();
  });
});
