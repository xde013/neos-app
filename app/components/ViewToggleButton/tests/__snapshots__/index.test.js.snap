// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewToggleButton /> Should render and match the snapshot 1`] = `
<button
  aria-pressed="true"
  class="MuiButtonBase-root MuiToggleButton-root Mui-selected"
  color="primary"
  tabindex="0"
  title="Toggle to Table View"
  type="button"
  value="false"
>
  <span
    class="MuiToggleButton-label"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h15c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 2v3H5V5h15zm-5 14h-5v-9h5v9zM5 10h3v9H5v-9zm12 9v-9h3v9h-3z"
      />
    </svg>
  </span>
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;
